src/Examine/LuceneEngine/Providers/LuceneIndex.cs

```diff
@@ -370,7 +392,7 @@ protected override void PerformIndexItems(IEnumerable<ValueSet> values, Action<I
         public void EnsureIndex(bool forceOverwrite)
         {
             if (!forceOverwrite && _exists.HasValue && _exists.Value) return;
-
+            if (_directory is ExamineDirectory examineDirectory && examineDirectory.IsReadOnly) return;
```

Comment:
if we keep this idea of ExamineDirectory then this check should be encapsulated in a protected property IsReadOnly since this same check `if (_directory is ExamineDirectory examineDirectory && examineDirectory.IsReadOnly) return;` is done a few times.

Reply:
@Shazwazza as pointed earlier I am not sure either about that :) but didnt removed that from my pr to this pr. :)